Lib/posixpath.py

```diff
@@ -439,10 +431,15 @@ def _joinrealpath(path, rest, strict, seen):
         if name == pardir:
             # parent dir
             if path:
-                path, name = split(path)
+                parent, name = split(path)
                 if name == pardir:
-                    path = join(path, pardir, pardir)
+                    # ../..
+                    path = join(path, pardir)
```

Comment:
Does this change semantics at all? Or is this safe? (@barneygale?)

Reply:
Note that in the old code, splitting the path had to be undone because path was overwritten. Here's some stub code:

```diff
-path, name = split(path)
-path = join(path, pardir) # undo split
+parent, name = split(path) # path isn't overwritten
path = join(path, pardir)
```

Reply:
I'm unresolving this because I wanted @barneygale to confirm your analysis. I'm not an expert in this code to the extent he is.

Reply:
This looks safe to me. The old code was popping a `..` segment off the end of the path to produce a dirname, and then adding two `..` segments back. The new code uses a different variable to store the dirname, and so only one `..` segment needs to be added.

Reply:
Fab -- thanks both :) excuse my paranoia!